modules/ipc/include/hephaestus/ipc/zenoh/action_server.h

```diff
@@ -0,0 +1,222 @@
+//================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <future>
+#include <optional>
+
+#include <magic_enum.hpp>
+
+#include "hephaestus/concurrency/message_queue_consumer.h"
+#include "hephaestus/ipc/common.h"
+#include "hephaestus/ipc/publisher.h"
+#include "hephaestus/ipc/zenoh/internal/action_server_client_helper.h"
+#include "hephaestus/ipc/zenoh/service.h"
+#include "hephaestus/ipc/zenoh/session.h"
+#include "hephaestus/ipc/zenoh/types/action_server_types.h"
+#include "hephaestus/ipc/zenoh/types/action_server_types_protobuf.h"  // NOLINT(misc-include-cleaner)
+
+namespace heph::ipc::zenoh {
+
+/// If an action server is already serving a request it will not accept new ones.
```

Comment:
It would be great to add a description what an action server does. If you think this is too much common knowledge, please make it brief.

ChatGPT suggestion:
```
In terms of Inter-Process Communication (IPC), an action server is a server that performs specific tasks in response to requests from clients. These tasks are usually well-defined actions that may take some time to complete, such as processing data, performing calculations, or interacting with hardware.

**Key Characteristics of an Action Server:**
**Asynchronous Processing:** Action servers typically handle requests asynchronously, allowing the client to continue its operations while waiting for the action to complete.
**Action Feedback:** While the action is in progress, the server can provide feedback to the client about the status or intermediate results.
**Result Handling:** Once the action is completed, the server sends the final result back to the client.
**State Management:** Action servers often maintain several states such as pending, active, preempted, succeeded, or aborted to manage the lifecycle of the tasks they perform. 
```